docs/make.jl

```diff
@@ -9,51 +9,53 @@ makedocs(
+        # You could read this end-to-end, or skip to what you need.
+        # Aim is to cover each new concept exactly once (but not list all variants).
+        # Hard to invent further divisions which aren't more confusing than helpful?
             "Quick Start" => "models/quickstart.md",
             "Fitting a Line" => "models/overview.md",
             "Gradients and Layers" => "models/basics.md",
-        ],
-        "Building Models" => [
-            "Built-in Layers 📚" => "models/layers.md",
+            "Training" => "training/training.md",
+            "Regularisation" => "models/regularisation.md",  # consolidated in #2114
             "Recurrence" => "models/recurrence.md",
-            "Activation Functions 📚" => "models/activation.md",
-            "NNlib.jl 📚 (`softmax`, `conv`, ...)" => "models/nnlib.md",
-         ],
-         "Handling Data" => [
-             "MLUtils.jl 📚 (`DataLoader`, ...)" => "data/mlutils.md",
-             "OneHotArrays.jl 📚 (`onehot`, ...)" => "data/onehot.md",
-         ],
-         "Training Models" => [
-             "Training" => "training/training.md",
-             "Regularisation" => "models/regularisation.md",
-             "Loss Functions 📚" => "models/losses.md",
-             "Optimisation Rules 📚" => "training/optimisers.md",  # TODO move optimiser intro up to Training
-             "Callback Helpers 📚" => "training/callbacks.md",
-             "Zygote.jl 📚 (`gradient`, ...)" => "training/zygote.md",
-         ],
-         "Model Tools" => [
-             "GPU Support" => "gpu.md",
-             "Saving & Loading" => "saving.md",
-             "Shape Inference 📚" => "outputsize.md",
-             "Weight Initialisation 📚" => "utilities.md",
-             "Flat vs. Nested 📚" => "destructure.md",
-             "Functors.jl 📚 (`fmap`, ...)" => "models/functors.md",
+            "GPU Support" => "gpu.md",
+            "Saving & Loading" => "saving.md",
+            "Performance Tips" => "performance.md",
+        ],
+        "Ecosystem" => "ecosystem.md",
+        "Reference" => [
+        # This essentially collects docstrings, with a bit of introduction.
+            "Built-in Layers" => "models/layers.md",
+            "Activation Functions" => "models/activation.md",
+            "Weight Initialisation" => "utilities.md",
+            "Loss Functions" => "models/losses.md",
+            "Optimisation Rules" => "training/optimisers.md",  # TODO move optimiser intro up to Training
+            "Shape Inference" => "outputsize.md",
+            "Flat vs. Nested" => "destructure.md",
+            "Callback Helpers" => "training/callbacks.md",
+            "NNlib.jl (`softmax`, `conv`, ...)" => "models/nnlib.md",
+            "Zygote.jl (`gradient`, ...)" => "training/zygote.md",
+            "MLUtils.jl (`DataLoader`, ...)" => "data/mlutils.md",
+            "Functors.jl (`fmap`, ...)" => "models/functors.md",
+            "OneHotArrays.jl (`onehot`, ...)" => "data/onehot.md",
```

Comment:
Having example functions right in the heading feels a little strange. Could we instead use adapt the tagline of each page? e.g. "Neural Network primitives - NNlib.jl".

Reply:
I don't love them, but they are compact, and I didn't manage to think of a better way (last time around). I mean I know what "primitives" means but it's not so obvious... what kind of utils is MLUtils, what on earth is Functors in less than a paragraph?

Reply:
My thought was to take inspiration from the shortened titles, something like so:
```suggestion
            "Low-level Operations - NNlib.jl" => "models/nnlib.md",
            "Taking Gradients - Zygote.jl" => "training/zygote.md",
            "Working with Data - MLUtils.jl" => "data/mlutils.md",
            "Manipulating Models - Functors.jl" => "models/functors.md",
            "One-Hot Encoding - OneHotArrays.jl" => "data/onehot.md",
```

Reply:
Last commit does this, see what you think (PR before + after):

<img width="470" alt="Screenshot 2022-12-07 at 20 04 48" src="https://user-images.githubusercontent.com/32575566/206330950-6c54f773-3221-487d-a169-cac640d4d607.png">

Reply:
Looks pretty good!

Reply:
Shall we do this, then refine as necc?